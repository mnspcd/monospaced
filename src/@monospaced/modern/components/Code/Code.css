.mdrn .Code {
  background-color: rgba(var(--color-grey-9-rgb), var(--opacity-static));
  border-radius: 0.25em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  padding: 0.25em 0.5em;
}

.mdrn .Code--onDarkBackground {
  background-color: rgba(
    var(--color-white-rgb),
    var(--opacity-static-inverted)
  );
}

.mdrn .Code--block {
  background-color: var(--color-cyan-8);
  border-radius: 0;
  color: var(--color-grey-3);
  display: block;
  font-weight: 300;
  line-height: var(--leading);
  -webkit-overflow-scrolling: touch;
  overflow-x: auto;
  padding: calc(1.5em / var(--type-scale-s)) calc(1em / var(--type-scale-s));
}

.mdrn .Code .hljs-comment,
.mdrn .Code .hljs-emphasis,
.mdrn .Code .hljs-quote {
  font-style: italic;
}

.mdrn .Code .hljs-strong {
  font-weight: 400; /* never bold */
}

.mdrn .Code .hljs-built_in,
.mdrn .Code .hljs-deletion {
  color: var(--color-red-4);
  font-weight: 400;
}

.mdrn .Code .hljs-name,
.mdrn .Code .hljs-section,
.mdrn .Code .hljs-selector-class,
.mdrn .Code .hljs-selector-id,
.mdrn .Code .hljs-title {
  color: var(--color-azure-4);
  font-weight: 400;
}

.mdrn .Code .hljs-doctag,
.mdrn .Code .hljs-literal,
.mdrn .Code .hljs-meta .hljs-meta-string,
.mdrn .Code .hljs-number,
.mdrn .Code .hljs-regexp,
.mdrn .Code .hljs-string {
  color: var(--color-cyan-4);
  font-weight: 400;
}

.mdrn .Code .hljs-addition,
.mdrn .Code .hljs-keyword,
.mdrn .Code .hljs-selector-tag {
  color: var(--color-green-4);
  font-weight: 400;
}

.mdrn .Code .hljs-attr,
.mdrn .Code .hljs-attribute,
.mdrn .Code .hljs-class .hljs-title,
.mdrn .Code .hljs-template-variable,
.mdrn .Code .hljs-type,
.mdrn .Code .hljs-variable {
  color: var(--color-yellow-4);
  font-weight: 400;
}

.mdrn .Code .hljs-bullet,
.mdrn .Code .hljs-link,
.mdrn .Code .hljs-meta .hljs-keyword,
.mdrn .Code .hljs-meta,
.mdrn .Code .hljs-selector-attr,
.mdrn .Code .hljs-selector-pseudo,
.mdrn .Code .hljs-subst,
.mdrn .Code .hljs-symbol {
  color: var(--color-orange-4);
  font-weight: 400;
}

/* Light */

.mdrn .Code--block.Code--lightMode {
  background-color: var(--color-orange-1);
  border-radius: 0;
  color: var(--color-grey-7);
  -webkit-font-smoothing: auto;
  -moz-osx-font-smoothing: auto;
}

.mdrn .Code--lightMode .hljs-built_in,
.mdrn .Code--lightMode .hljs-deletion {
  color: var(--color-red-6);
}

.mdrn .Code--lightMode .hljs-name,
.mdrn .Code--lightMode .hljs-section,
.mdrn .Code--lightMode .hljs-selector-class,
.mdrn .Code--lightMode .hljs-selector-id,
.mdrn .Code--lightMode .hljs-title {
  color: var(--color-azure-6);
}

.mdrn .Code--lightMode .hljs-doctag,
.mdrn .Code--lightMode .hljs-literal,
.mdrn .Code--lightMode .hljs-meta .hljs-meta-string,
.mdrn .Code--lightMode .hljs-number,
.mdrn .Code--lightMode .hljs-regexp,
.mdrn .Code--lightMode .hljs-string {
  color: var(--color-cyan-6);
}

.mdrn .Code--lightMode .hljs-addition,
.mdrn .Code--lightMode .hljs-keyword,
.mdrn .Code--lightMode .hljs-selector-tag {
  color: var(--color-green-6);
}

.mdrn .Code--lightMode .hljs-attr,
.mdrn .Code--lightMode .hljs-attribute,
.mdrn .Code--lightMode .hljs-class .hljs-title,
.mdrn .Code--lightMode .hljs-template-variable,
.mdrn .Code--lightMode .hljs-type,
.mdrn .Code--lightMode .hljs-variable {
  color: var(--color-yellow-6);
}

.mdrn .Code--lightMode .hljs-bullet,
.mdrn .Code--lightMode .hljs-link,
.mdrn .Code--lightMode .hljs-meta .hljs-keyword,
.mdrn .Code--lightMode .hljs-meta,
.mdrn .Code--lightMode .hljs-selector-attr,
.mdrn .Code--lightMode .hljs-selector-pseudo,
.mdrn .Code--lightMode .hljs-subst,
.mdrn .Code--lightMode .hljs-symbol {
  color: var(--color-orange-6);
}
