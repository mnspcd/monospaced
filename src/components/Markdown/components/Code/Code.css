.Code {
  background-color: rgba(var(--mnspcd-color-grey-9-rgb), var(--opacity-static));
  border-radius: 0.25em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  padding: 0.25em 0.5em;
}

.Code--onDarkBackground {
  background-color: rgba(var(--mnspcd-color-white-rgb), var(--opacity-static-inverted));
}

.Code--block {
  background-color: var(--mnspcd-color-cyan-8);
  border-radius: 0;
  color: var(--mnspcd-color-grey-3);
  display: block;
  font-weight: 300;
  line-height: var(--leading);
  -webkit-overflow-scrolling: touch;
  overflow-x: auto;
  padding: calc(1.5em / var(--type-scale-s)) calc(1em / var(--type-scale-s));
}

.Code .hljs-comment,
.Code .hljs-emphasis,
.Code .hljs-quote {
  font-style: italic;
}

.Code .hljs-strong {
  font-weight: 400; /* never bold */
}

.Code .hljs-built_in,
.Code .hljs-deletion {
  color: var(--mnspcd-color-red-4);
  font-weight: 400;
}

.Code .hljs-name,
.Code .hljs-section,
.Code .hljs-selector-class,
.Code .hljs-selector-id,
.Code .hljs-title {
  color: var(--mnspcd-color-azure-4);
  font-weight: 400;
}

.Code .hljs-doctag,
.Code .hljs-literal,
.Code .hljs-meta .hljs-meta-string,
.Code .hljs-number,
.Code .hljs-regexp,
.Code .hljs-string {
  color: var(--mnspcd-color-cyan-4);
  font-weight: 400;
}

.Code .hljs-addition,
.Code .hljs-keyword,
.Code .hljs-selector-tag {
  color: var(--mnspcd-color-green-4);
  font-weight: 400;
}

.Code .hljs-attr,
.Code .hljs-attribute,
.Code .hljs-class .hljs-title,
.Code .hljs-template-variable,
.Code .hljs-type,
.Code .hljs-variable {
  color: var(--mnspcd-color-yellow-4);
  font-weight: 400;
}

.Code .hljs-bullet,
.Code .hljs-link,
.Code .hljs-meta .hljs-keyword,
.Code .hljs-meta,
.Code .hljs-selector-attr,
.Code .hljs-selector-pseudo,
.Code .hljs-subst,
.Code .hljs-symbol {
  color: var(--mnspcd-color-orange-4);
  font-weight: 400;
}

/* Light */

.Code--block.Code--lightMode {
  background-color: var(--mnspcd-color-orange-1);
  border-radius: 0;
  color: var(--mnspcd-color-grey-7);
  -webkit-font-smoothing: auto;
  -moz-osx-font-smoothing: auto;
}

.Code--lightMode .hljs-built_in,
.Code--lightMode .hljs-deletion {
  color: var(--mnspcd-color-red-6);
}

.Code--lightMode .hljs-name,
.Code--lightMode .hljs-section,
.Code--lightMode .hljs-selector-class,
.Code--lightMode .hljs-selector-id,
.Code--lightMode .hljs-title {
  color: var(--mnspcd-color-azure-6);
}

.Code--lightMode .hljs-doctag,
.Code--lightMode .hljs-literal,
.Code--lightMode .hljs-meta .hljs-meta-string,
.Code--lightMode .hljs-number,
.Code--lightMode .hljs-regexp,
.Code--lightMode .hljs-string {
  color: var(--mnspcd-color-cyan-6);
}

.Code--lightMode .hljs-addition,
.Code--lightMode .hljs-keyword,
.Code--lightMode .hljs-selector-tag {
  color: var(--mnspcd-color-green-6);
}

.Code--lightMode .hljs-attr,
.Code--lightMode .hljs-attribute,
.Code--lightMode .hljs-class .hljs-title,
.Code--lightMode .hljs-template-variable,
.Code--lightMode .hljs-type,
.Code--lightMode .hljs-variable {
  color: var(--mnspcd-color-yellow-6);
}

.Code--lightMode .hljs-bullet,
.Code--lightMode .hljs-link,
.Code--lightMode .hljs-meta .hljs-keyword,
.Code--lightMode .hljs-meta,
.Code--lightMode .hljs-selector-attr,
.Code--lightMode .hljs-selector-pseudo,
.Code--lightMode .hljs-subst,
.Code--lightMode .hljs-symbol {
  color: var(--mnspcd-color-orange-6);
}
