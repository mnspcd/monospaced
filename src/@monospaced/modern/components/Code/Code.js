import classNames from "classnames";
import PropTypes from "prop-types";
import React, { useContext } from "react";
import SyntaxHighlighter from "react-syntax-highlighter";

import { SurfaceContext } from "../Surface";

import "./Code.css";

/**
 * Use `Code` to display code samples. As inline text, or code blocks with
 * optional syntax highlighting.
 */
const Code = ({ block = false, children, lightMode = false }) => {
  const surfaceBackground = useContext(SurfaceContext);

  return (
    <code
      className={classNames({
        Code: true,
        [`Code--block`]: block,
        [`Code--lightMode`]: lightMode,
        [`Code--onDarkBackground`]: surfaceBackground === "dark",
      })}
    >
      {children}
    </code>
  );
};

Code.propTypes = {
  /**
   * Block variant of Code.
   */
  block: PropTypes.bool,

  /**
   * Content to display in Code. Either preformatted text, or syntax-highlighted
   * markup in highlight.js format. MDX code language blocks rendered by
   * Code.Block will be syntax-highlighted automatically.
   */
  children: PropTypes.node.isRequired,

  /**
   * Light variant of Code. Only applies if block is true.
   */
  lightMode: PropTypes.bool,
};

Code.Block = ({ children, className, lightMode }) => (
  <Code block lightMode={lightMode}>
    <HighlightedChildren language={className}>{children}</HighlightedChildren>
  </Code>
);

Code.Block.displayName = "Code.Block";

Code.Block.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  lightMode: PropTypes.bool,
};

/* eslint-disable react/display-name, react/prop-types */

Code.Block.LightMode = ({ children, className }) => (
  <Code block lightMode>
    <HighlightedChildren language={className}>{children}</HighlightedChildren>
  </Code>
);

Code.Block.LightMode.displayName = "Code.Block.LightMode";

Code.Block.LightMode.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

/* eslint-enable */

const HighlightedChildren = ({ children, language }) => (
  <SyntaxHighlighter
    CodeTag="span"
    language={language ? `${language.replace(/language-/, "")}` : "text"}
    PreTag="span"
    useInlineStyles={false}
  >
    {children}
  </SyntaxHighlighter>
);

HighlightedChildren.propTypes = {
  children: PropTypes.node.isRequired,
  language: PropTypes.string,
};

export default Code;
