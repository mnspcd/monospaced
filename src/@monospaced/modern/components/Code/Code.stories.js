import { MDXProvider } from "@mdx-js/react";
import { boolean, select, text } from "@storybook/addon-knobs";
import React from "react";

import Surface from "../Surface";

import Code from "./Code";
import MdxMockJsCode from "./__mocks__/code.js.mock.mdx";

export default {
  component: Code,
  title: "Components/Code",
};

export const basic = () => <Code>const foo = 10</Code>;

export const onDarkBackground = () => (
  <Surface backgroundColor="grey-dark">
    <div style={{ padding: "1em" }}>
      <Code>const foo = 10</Code>
    </div>
  </Surface>
);

export const block = () => (
  <pre>
    <Code block>const foo = 10</Code>
  </pre>
);

export const highlighted = () => (
  <MDXProvider components={{ code: Code.Block }}>
    <MdxMockJsCode />
  </MDXProvider>
);

export const blockLightMode = () => (
  <pre>
    <Code block lightMode>
      const foo = 10
    </Code>
  </pre>
);

export const highlightedLightMode = () => (
  <MDXProvider components={{ code: Code.Block.LightMode }}>
    <MdxMockJsCode />
  </MDXProvider>
);
export const knobs = () => {
  const code = text("text", "const foo = 10", "Code");

  return (
    <Surface
      backgroundColor={select(
        "backgroundColor",
        Surface.backgroundColors,
        "white",
        "Surface",
      )}
    >
      <Code
        block={boolean("block", false, "Code")}
        lightMode={boolean("lightMode", false, "Code")}
      >
        {code}
      </Code>
    </Surface>
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
