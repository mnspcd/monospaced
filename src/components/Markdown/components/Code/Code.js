import PropTypes from "prop-types";
import React from "react";
import SyntaxHighlighter from "react-syntax-highlighter";
import classNames from "classnames";

import { SurfaceConsumer } from "../../../Surface";
import "./Code.css";

const Code = ({ block, children, lightMode }) => (
  <SurfaceConsumer>
    {surfaceBackground => (
      <code
        className={classNames({
          Code: true,
          [`Code--block`]: block,
          [`Code--lightMode`]: lightMode,
          [`Code--onDarkBackground`]: surfaceBackground === "dark",
        })}
      >
        {children}
      </code>
    )}
  </SurfaceConsumer>
);

Code.propTypes = {
  block: PropTypes.bool,
  children: PropTypes.node.isRequired,
  lightMode: PropTypes.bool,
};

Code.Block = ({ children, className, lightMode }) => (
  <Code block lightMode={lightMode}>
    <HighlightedChildren language={className}>{children}</HighlightedChildren>
  </Code>
);

Code.Block.displayName = "Code.Block";

Code.Block.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  lightMode: PropTypes.bool,
};

/* eslint-disable react/display-name, react/prop-types */

Code.Block.LightMode = ({ children, className }) => (
  <Code block lightMode>
    <HighlightedChildren language={className}>{children}</HighlightedChildren>
  </Code>
);

Code.Block.LightMode.displayName = "Code.Block.LightMode";

Code.Block.LightMode.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

/* eslint-enable */

const HighlightedChildren = ({ children, language }) => (
  <SyntaxHighlighter
    CodeTag="span"
    language={language ? `${language.replace(/language-/, "")}` : "text"}
    PreTag="span"
    useInlineStyles={false}
  >
    {children}
  </SyntaxHighlighter>
);

HighlightedChildren.propTypes = {
  children: PropTypes.node.isRequired,
  language: PropTypes.string,
};

export default Code;
