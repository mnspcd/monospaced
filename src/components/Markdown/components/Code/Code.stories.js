import { MDXProvider } from "@mdx-js/react";
import React from "react";
import { boolean, select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Code from "./Code";
import Surface from "../../../Surface";
import MdxMockJsCode from "./__mocks__/code.js.mock.mdx";

storiesOf("Components/Markdown/Code", module)
  .add("Block", () => (
    <pre>
      <Code lightMode={boolean("lightMode", false)} block>
        {text("text", 'const foo = "bar";')}
      </Code>
    </pre>
  ))
  .add("Highlighted", () => {
    return boolean("lightMode", false) ? (
      <MDXProvider components={{ code: Code.Block.LightMode }}>
        <MdxMockJsCode />
      </MDXProvider>
    ) : (
      <MDXProvider components={{ code: Code.Block }}>
        <MdxMockJsCode />
      </MDXProvider>
    );
  })
  .add("Inline", () => {
    const code = text("text", 'const foo = "bar";', "Code");

    return (
      <Surface
        backgroundColor={select(
          "backgroundColor",
          Surface.backgroundColors,
          "white",
          "Surface",
        )}
      >
        <Code>{code}</Code>
      </Surface>
    );
  });
