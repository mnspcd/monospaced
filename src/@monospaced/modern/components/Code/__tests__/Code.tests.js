/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Code from "..";

describe("Code component", () => {
  it("should render correctly", () => {
    const { container } = render(<Code>children</Code>);

    expect(container).toMatchSnapshot();
  });

  it("should render a code block correctly", () => {
    const { container } = render(<Code.Block>children</Code.Block>);

    expect(container).toMatchSnapshot();
  });

  it("should render a code block correctly in light mode", () => {
    const { container } = render(
      <Code.Block.LightMode>children</Code.Block.LightMode>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render a code language correctly", () => {
    const { container } = render(
      <Code.Block className="language-js">children</Code.Block>,
    );

    expect(container).toMatchSnapshot();
  });
});
