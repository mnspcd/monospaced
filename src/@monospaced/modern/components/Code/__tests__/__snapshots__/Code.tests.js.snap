// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code component should render a code block correctly 1`] = `
<code
  class="Code Code--block"
>
  <span
    class="hljs"
  >
    <span>
      children
    </span>
  </span>
</code>
`;

exports[`Code component should render a code block correctly in light mode 1`] = `
<code
  class="Code Code--block Code--lightMode"
>
  <span
    class="hljs"
  >
    <span>
      children
    </span>
  </span>
</code>
`;

exports[`Code component should render a code language correctly 1`] = `
<code
  class="Code Code--block"
>
  <span
    class="hljs"
  >
    <span>
      children
    </span>
  </span>
</code>
`;

exports[`Code component should render correctly 1`] = `
<code
  class="Code"
>
  children
</code>
`;
